core/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala

```diff
@@ -775,6 +781,22 @@ object SQLPlanParser extends Logging {
     parsedExpressions.distinct.toArray
   }
 
+  def parseWindowGroupLimitExpression(exprStr: String): Option[String] = {
+    // [category#16], [amount#17 DESC NULLS LAST], dense_rank(amount#17), 2, Final
+
+    // This splits the string to get only the ranking expression in WindowGroupLimitExec.
+    // So we first split the string on comma and get the third element from the array.
+    // dense_rank(amount#17)
+    val rankLikeExpr = exprStr.split(", ").lift(2).map(_.trim)
+    rankLikeExpr.flatMap { rankExpr =>
+      // Get function name from WindowExpression
+      val rankLikeFunc = windowFunctionPattern.findFirstIn(rankExpr)
+      val functionName = rankLikeFunc.flatMap(getFunctionName(windowFunctionPattern, _))
+      // Validate if the function is a supported rank like function.
+      functionName.filter(supportedWindowGroupLimitFunctions.contains)
```

Comment:
Q: Why do we filter expressions in this method?
It is better if the parser method purpose to be limited to generate the list of expressions.
Then the logic of deciding how to decide if the SQL is supported or not should be in the `WindowGroupLimitParser` class itself.